frontend: Add option to "spin off" subqueries as actual range queries (#10460)
grafana/mimir: docs/sources/mimir/configure/configuration-parameters/index.md

# CLI flag: -query-frontend.use-active-series-decoder
[use_active_series_decoder: <boolean> | default = false]

> # (experimental) If set, subqueries in instant queries will be spun off as range

Reviewer: ```suggestion
# (experimental) If set, subqueries in instant queries are spun off as range
```

> # queries and sent to the given URL. Also requires
> # `instant_queries_with_subquery_spin_off` to be set for the tenant.

Reviewer: ```suggestion
# queries and sent to the given URL. This parameter also requires you to set
# `instant_queries_with_subquery_spin_off` for the tenant.
```

# CLI flag: -query-frontend.prom2-range-compat
[prom2_range_compat: <boolean> | default = false]

> # (experimental) List of regexp patterns matching instant queries. Subqueries

Reviewer: ```suggestion
# (experimental) List of regular expression patterns matching instant queries. Subqueries
```

> # within those instant queries will be spun off as range queries to optimize

Reviewer: ```suggestion
# within those instant queries are spun off as range queries to optimize
```

> # their execution.

Reviewer: Could we say `performance` instead of `execution`?
